gpflow/conditionals.py

```diff
@@ -124,6 +125,68 @@ def base_conditional(Kmn, Kmm, Knn, f, full_cov=False, q_sqrt=None, whiten=False
     return fmean, fvar
 
 
+@name_scope()
+def uncertain_conditional(Xnew_mu, Xnew_var, feat, kern, q_mu, q_sqrt, whiten=False):
+    """
+    Calculates the conditional for uncertain inputs Xnew, p(Xnew) = N(Xnew_mu, Xnew_var).
+    See ``conditional`` documentation for further reference.
+
+    :param Xnew_mu: mean of the inputs, size N x D
+    :param Xnew_var: covariance matrix of the inputs, size N x D x D
```

Comment:
Perhaps a bit of pain to implement, but it would be nice to support diag Xnew_var